hack/validate/shfmt

```diff
@@ -2,9 +2,7 @@
 set -e -o pipefail
 
 shfmtflags=(-bn -ci -sr -d "$@") # Allows to specify additional flags.
-# NOTE: `git grep '^#!'` may also pick up non-shell script files.
-# Add exceptional files to `grep -v` if any false-positive is detected.
-if git grep --name-only '^#!' | grep -E -v '(vendor|\.go|Jenkinsfile|\.py|\.bats)' \
+if shfmt -f . | grep -E -v '^vendor/' \
 	| xargs shfmt "${shfmtflags[@]}"; then
```

Comment:
Curious; if `shfmt` is able to find them all, does it have an option to also format them all (without having to down the `xargs` road)?

Oh, I guess it may not have an option to exclude paths?

Reply:
It looks like `shfmt -f` is a separate mode of operation. 

But, we can add `.editorconfig` file (which shfmt understands) to exclude vendor, similar to https://github.com/opencontainers/runc/pull/3379.

Reply:
Well, I wanted to add `.editorconfig` to avoid using `shfmt -f | grep -v ... | shfmt` pipe, but it was added to `.gitignore` by #27490, so I'm unsure whether should I revert that change; maybe some people do have `.editorconfig` locally, and thus the added one will disrupt their workflow.

Reply:
So, if you want to reconsider what was decided in #27490, I can add .editorconfig and make shfmt run as a single command.